app/models/spree/gift_card_barcode.rb

```diff
@@ -0,0 +1,47 @@
+module Spree
+  class GiftCardBarcode < Spree::Base
+    acts_as_paranoid
+    validates_uniqueness_of :number, unless: :deleted_at
+
+    belongs_to :virtual_gift_card, class_name: 'Spree::VirtualGiftCard'
+    belongs_to :order, class_name: 'Spree::Order'
+
+    scope :unredeemed, -> { where(redeemed_at: nil) }
+
+    self.whitelisted_ransackable_attributes = %w[number]
+
+    def sold!(gift_card, order)
+      self.order = order
+      self.virtual_gift_card = gift_card
+      self.sold_at = Time.now
+
+      save
+    end
+
+    def redeem(user)
```

Comment:
When will we use this versus going up to the virtual gift card and calling redeem?